include/secp256k1_ecdsa_adaptor.h

```diff
@@ -0,0 +1,156 @@
+#ifndef SECP256K1_ECDSA_ADAPTOR_H
+#define SECP256K1_ECDSA_ADAPTOR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** This module implements single signer ECDSA adaptor signatures following
+ *  "One-Time Verifiably Encrypted Signatures A.K.A. Adaptor Signatures" by
+ *  Lloyd Fournier
+ *  (https://lists.linuxfoundation.org/pipermail/lightning-dev/2019-November/002316.html
+ *  and https://github.com/LLFourn/one-time-VES/blob/master/main.pdf).
+ *
+ *  TODO: remove disclaimer
+ *  Note that at this module is currently a work in progress. It's not secure
+ *  nor stable. Let me repeat: IT IS EXTREMELY DANGEROUS AND RECKLESS TO USE
+ *  THIS MODULE IN PRODUCTION. DON'T!
+*/
+
+/** A pointer to a function to deterministically generate a nonce.
+ *
+ *  Same as secp256k1_nonce_function_hardened with the exception of using the
+ *  compressed 33-byte encoding for the pubkey argument.
+ *
+ *  Returns: 1 if a nonce was successfully generated. 0 will cause signing to
+ *           return an error.
+ *  Out:     nonce32:   pointer to a 32-byte array to be filled by the function.
+ *  In:        msg32:   the 32-byte message hash being verified (will not be null)
+ *             key32:   pointer to a 32-byte secret key (will not be null)
+ *              pk33:   the 33-byte serialized pubkey corresponding to key32
+ *                      (will not be null)
+ *            algo16:   pointer to a 16-byte array describing the signature
+ *                      algorithm (will not be null).
```

Comment:
The [schnorrsig API overhaul PR](https://github.com/bitcoin-core/secp256k1/pull/844) adds an `algolen` argument to the bip340 nonce function which gets rid of the NUL byte weirdness (initially we didn't add arguments because we wanted to have compatibility with the original nonce function). I think it would be best to do the same thing here.